### PR TITLE
Add table `aws_appconfig_application` Closes #1238

### LINKED ISSUE #1238: Add table `aws_appconfig_application`
**References**
TF Ref - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_application

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

description = "Example AppConfig Application"
id = "tr21909"
resource_aka = "arn:aws:appconfig:us-east-1:12345678910:application/tr21909"
resource_name = "turbottest31890"
tags_all = tomap({
  "Type" = "AppConfig Application"
})

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:appconfig:us-east-1:12345678910:application/tr21909",
    "description": "Example AppConfig Application",
    "id": "tr21909",
    "name": "turbottest31890",
    "tags": {
      "Type": "AppConfig Application"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:appconfig:us-east-1:12345678910:application/tr21909",
    "description": "Example AppConfig Application",
    "id": "tr21909",
    "name": "turbottest31890",
    "tags": {
      "Type": "AppConfig Application"
    }
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:appconfig:us-east-1:12345678910:application/tr21909"
    ],
    "tags": {
      "Type": "AppConfig Application"
    },
    "title": "turbottest31890"
  }
]
✔ PASSED

POSTTEST: tests/aws_appconfig_application

TEARDOWN: tests/aws_appconfig_application

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select
  arn,
  id,
  name,
  description,
  tags
from
  aws_appconfig_application;
+--------------------------------------------------------------+---------+-----------------+-------------------------------+----------------------------------+
| arn                                                          | id      | name            | description                   | tags                             |
+--------------------------------------------------------------+---------+-----------------+-------------------------------+----------------------------------+
| arn:aws:appconfig:us-east-1:111122223333:application/ikjegeg | ikjegeg | turbottest21720 | Example AppConfig Application | {"Type":"AppConfig Application"} |
+--------------------------------------------------------------+---------+-----------------+-------------------------------+----------------------------------+
```
</details>

### CODE REVIEW

File: aws/service_v2.go [hunk 1]
```diff
@@ -64,6 +65,14 @@ func APIGatewayV2Client(ctx context.Context, d *plugin.QueryData) (*apigatewayv2
 	return apigatewayv2.NewFromConfig(*cfg), nil
 }
 
+func AppConfigService(ctx context.Context, d *plugin.QueryData) (*appconfig.Client, error) {
```
Comment (ParthaI) [inline aws/service_v2.go:68]:
```suggestion
func AppConfigClient(ctx context.Context, d *plugin.QueryData) (*appconfig.Client, error) {
```

File: aws/table_aws_appconfig_application.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	}
+}
+
+//// LIST FUNCTION
+
+func listAppConfigApplication(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	// Create Session
+	svc, err := AppConfigService(ctx, d)
```
Comment (ParthaI) [inline aws/table_aws_appconfig_application.go:83]:
```suggestion
	svc, err := AppConfigClient(ctx, d)
```

File: aws/table_aws_appconfig_application.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAppConfigApplication(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	// Create Session
+	svc, err := AppConfigService(ctx, d)
```
Comment (ParthaI) [inline aws/table_aws_appconfig_application.go:134]:
```suggestion
	svc, err := AppConfigClient(ctx, d)
```

File: aws/table_aws_appconfig_application.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	}
+
+	return nil, nil
+}
+
+func getAppConfigTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	// Create Session
+	svc, err := AppConfigService(ctx, d)
```
Comment (ParthaI) [inline aws/table_aws_appconfig_application.go:167]:
```suggestion
	svc, err := AppConfigClient(ctx, d)
```

**Review by khushboo9024 — APPROVED** [review]

File: docs/tables/aws_appconfig_application.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Table: aws_appconfig_application
+
+In AWS AppConfig, an application is simply an organizational construct like a folder. This organizational construct has a relationship with some unit of executable code.
+
+## Examples
+
+### Basic info
```
Comment (cbruno10) [inline docs/tables/aws_appconfig_application.md:7]: Based off of the lack of examples and useful columns, it's hard to tell why this table is required. @karanpopat @khushboo9024 Do you know why the request came up to create this table? Is it meant to join with another table's data? Do we need other AppConfig tables for this table to be useful?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see review comments, thanks!

File: docs/tables/aws_appconfig_application.md (hunk 5 above)
Comment (cbruno10) [reply docs/tables/aws_appconfig_application.md:7]: While this table by itself is not very useful, later we will add tables like `aws_appconfig_environment`, which will use this table as a parent hydrate. So we'll add this table for now and then add additional AppConfig tables later on.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**